src/proto.ts

```diff
@@ -73,7 +73,8 @@ export function jsonTemplate(message: protobuf.Type): any {
     }
     if (
       field.resolvedType !== null &&
-      field.resolvedType instanceof protobuf.Type
+      field.resolvedType instanceof protobuf.Type &&
+      field.type !== 'Struct' // the Struct type can have nested structs, which causes infinite recursion
```

Comment:
Does this need to also check the namespace?

Reply:
Hm not sure if I have access to namespace from here, but I can imagine it causing issues if someone created a message named `Struct` 😅

Reply:
So is the issue that a message can embed itself?

Reply:
Yep - but not directly. Struct has a `map` which contains [Values](https://developers.google.com/protocol-buffers/docs/reference/google.protobuf#google.protobuf.Value), which can be a `Struct`.